File: problems/algorithms/namespace-array-util-object/php/code/spec/ArrayUtil.spec.php
<?php
require __DIR__ . '/../src/ArrayUtil.php';

describe('Array Util', function () {
  it('finding lowest value in [1, 4, 2, 6, 10, 3]', function () {
    expect(ArrayUtil::min([1, 4, 2, 6, 10, 3]))->toBe(1.0);
  });

  it('finding lowest value in [1, 4, -1, 6, 10, 3]', function () {
    expect(ArrayUtil::min([1, 4, -1, 6, 10, 3]))->toBe(-1.0);
  });

  it('finding greatest value in [1, 4, 2, 6, 10, 3]', function () {
    expect(ArrayUtil::max([1, 4, 2, 6, 10, 3]))->toBe(10.0);
  });

  it('generating range of numbers from 0 to 10', function () {
    expect(ArrayUtil::range(10))->toEqual([0, 1, 2, 3, 4, 5, 6, 7, 8, 9]);
  });

  it('generating range of numbers from 1 to 11', function () {
    expect(ArrayUtil::range(1, 11))->toEqual([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
  });

  it('generating range of numbers from 0 to 10 with steps', function () {
    expect(ArrayUtil::range(0, 30, 5))->toEqual([0, 5, 10, 15, 20, 25]);
  });

  it("generating zip of ['moe', 'larry'] and [30, 40]", function () {
    expect(ArrayUtil::zip(['moe', 'larry'], [30, 40]))->toEqual([['moe', 30], ['larry', 40]]);
  });

  it("generating zip of ['moe', 'larry', 'curly'], [30, 40, 50] and [true, false, false]", function () {
    expect(ArrayUtil::zip(['moe', 'larry', 'curly'], [30, 40, 50], [true, false, false]))->toEqual([
      ['moe', 30, true],
      ['larry', 40, false],
      ['curly', 50, false],
    ]);
  });

  it('removing duplicate values in [1, 2, 1, 4, 1, 3]', function () {
    expect(ArrayUtil::uniq([1, 2, 1, 4, 1, 3]))->toEqual([1, 2, 4, 3]);
  });

  it('removing duplicate values in [1, 2, 1, 3, 3]', function () {
    expect(ArrayUtil::uniq([1, 2, 1, 3, 3]))->toEqual([1, 2, 3]);
  });
});

File: src/pages/exercises/array-util/_codes/ruby/response/spec/array_util.spec.rb
require_relative "../src/array_util"

describe ArrayUtil do
  describe ".array_util" do

    context "min" do
      it "return 3" do
        expect(ArrayUtil.min([1, 4, 2, 6, 10, 3])).to eql(1)
      end
    end

    context "max" do
      it "return 10" do
        expect(ArrayUtil.max([1, 4, 2, 6, 10, 3])).to eql(10)
      end
    end

    context "range" do
      it "return 0, 1, 2, 3, 4, 5, 6, 7, 8 ,9" do
        expect(ArrayUtil.range(10)).to eql([0, 1, 2, 3, 4, 5, 6, 7, 8, 9])
      end
    end

    context "zip" do
      it "return [['moe', 30], ['larry', 40]]" do
        expect(ArrayUtil.zip(['moe', 'larry'], [30, 40])).to eql([['moe', 30], ['larry', 40]])
      end
    end

    context "zip" do
      it "return ['moe', 30, true], ['larry', 40, false], ['curly', 50, false]" do
        expect(ArrayUtil.zip(['moe', 'larry', 'curly'], [30, 40, 50], [true, false, false]))
        .to eql([
          ['moe', 30, true], ['larry', 40, false],['curly', 50, false],
        ])
      end
    end

    context "uniq" do
      it "return [1, 2, 3]" do
        expect(ArrayUtil.uniq([1, 2, 1, 3, 3])).to eql([1, 2, 3])
      end
    end

    context "sortNum" do
      it "return [1, 2, 3, 10, 32]" do
        expect(ArrayUtil.sortNum([1, 2, 10, 3, 32])).to eql([1, 2, 3, 10, 32])
      end
    end

  end
end
 